platforms/documentation/docs/src/docs/userguide/optimizing-performance/gradle_daemon.adoc

```diff
@@ -171,43 +171,60 @@ You can also kill Daemons manually with your operating system.
+NOTE: Daemon JVM discovery and criteria are <<feature_lifecycle.adoc#sec:incubating_state,incubating>> features and are subject to change in a future release.
 
-By default, the same JVM installation that starts the build is used to run the build.
-Gradle uses the current shell path and `JAVA_HOME` environment variable to locate a usable JVM.
+By default, the Gradle daemon runs on the same JVM installation that started the build.
+Gradle defaults to the current shell path and `JAVA_HOME` environment variable to locate a usable JVM.
 
-You can also specify a different JVM installation for the build using the <<build_environment.adoc#sec:gradle_configuration_properties,`org.gradle.java.home`>> property or programmatically through the Tooling API.
+Alternatively, you could specify a different JVM installation for the build using the <<build_environment.adoc#sec:gradle_configuration_properties,`org.gradle.java.home` Gradle property>> or programmatically through the Tooling API.
 
-Using the _Daemon JVM discovery_ and _Daemon JVM criteria_ Gradle records the build's JVM requirements.
-This controls the JVM used to run Gradle itself.
+Building on the <<toolchains.adoc#toolchains,toolchain feature>>, you can now use declarative criteria to specify the JVM requirements for the build.
 
 === Daemon JVM criteria
 
-The _Daemon JVM criteria_ feature is similar to the `wrapper` task, which captures the current Gradle version.
-It uses the built-in `updateDaemonJvm` task to record the JVM requirements for the build based on the current JVM installation.
+The _daemon JVM criteria_ feature is built based on a task, similar to the <<gradle_wrapper.adoc#customizing_wrapper,`wrapper` one>>.
+The task uses its configuration in the build script or command-line arguments to capture a JVM definition in the `gradle/daemon-jvm.properties` file.
 
-NOTE: Currently, Gradle only considers the major JVM version when checking compatibility.
+As with the wrapper, the generated file should be checked into version control.
+This will ensure any developer or CI server that runs the build will use the same JVM version.
 
-=== Daemon JVM discovery
+With the following configuration, running `./gradlew updateDaemonJvm` will generate the file shown below:
+====
+include::sample[dir="snippets/daemon-jvm/customized-task/kotlin",files="build.gradle.kts[tags=customized-daemon-jvm-task]"]
+include::sample[dir="snippets/daemon-jvm/customized-task/groovy",files="build.gradle[tags=customized-daemon-jvm-task]"]
+====
 
-On subsequent builds, _Daemon JVM discovery_ uses the recorded JVM requirements to locate a compatible JVM installation using the same underlying mechanisms as <<toolchains.adoc#toolchains, Java Toolchains>>.
-This ensures every person who starts the build will use the same JVM version.
+====
+.gradle/daemon-jvm.properties
+[source,properties]
+----
+#This file is generated by updateDaemonJvm
```

Comment:
```suggestion
# This file is generated by updateDaemonJvm
```

Reply:
Yes, we should fix that, but that's a code change. I copied it from a real generated task